File: cpp/238_ProductofArrayExceptSelf.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int len = nums.size();
        vector<int> ret(len, 1);
        for(int i = 1; i < len; ++i){
            ret[i] = ret[i-1] * nums[i-1];
        }
        int temp = 1;
        for(int i = len-1; i >= 0; --i){
            ret[i] *= temp;
            temp *= nums[i];
        }
        return ret;
    }
};

File: js/238_ProductofArrayExceptSelf.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
    var ret = [];
    var len = nums.length;
    for(var n = 0; n < len; ++n){
        ret.push(1);
    }
    
    for(var i = 1; i < len; ++i){
        ret[i] = ret[i-1] * nums[i-1];
    }
    var temp = 1;
    for(var j = len-1; j >= 0; --j){
        ret[j] *= temp;
        temp *= nums[j];
    }
    return ret;
};